### PR TITLE
[move][move-2024] Add support for inline module definitions

### PR DESCRIPTION
**Pull request by cgswords**
## Description 

Exactly what it says on the tin.

## Test plan 

New tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: external-crates/move/crates/move-compiler/src/editions/mod.rs [hunk 1]
```diff
@@ -279,6 +281,7 @@ impl FeatureGate {
             FeatureGate::NoParensCast => "'as' without parentheses is",
             FeatureGate::TypeHoles => "'_' placeholders for type inference are",
             FeatureGate::Lambda => "lambda expressions are",
+            FeatureGate::InlineModule => "inline 'module' forms (ending with ';') are",
```
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/src/editions/mod.rs:284]: Is inline the right name here? 
Not sure I have a better one, but maybe something like 
`"'module' label forms (ending with ';') are"`

Comment (tzakian) [reply external-crates/move/crates/move-compiler/src/editions/mod.rs:284]: +1 to the naming. What if we just call them "module labels" or something similar?

**Review by tnowacki — APPROVED** [review]

File: external-crates/move/crates/move-compiler/tests/move_2024/parser/top_level_module_double_invalid.exp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+error[E02004]: invalid 'module' declaration
+  ┌─ tests/move_2024/parser/top_level_module_double_invalid.move:7:11
+  │
+7 │ module a::n;
+  │           ^ Cannot define a 'module' label form in a file with multiple modules
```
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/tests/move_2024/parser/top_level_module_double_invalid.exp:5]: Possible to point to the other module? Or too difficult in parsing?

Comment (cgswords) [reply external-crates/move/crates/move-compiler/tests/move_2024/parser/top_level_module_double_invalid.exp:8]: Done and done

**Pull request merged**